cobalt/media/base/metrics_provider.cc

```diff
@@ -122,6 +122,17 @@ void MediaMetricsProvider::EndTrackingAction(MediaAction action) {
   tracked_actions_start_times_.erase(action);
 }
 
+void MediaMetricsProvider::EndTrackingAction(MediaAction action,
+                                             int number_of_samples) {
+  DCHECK(IsActionCurrentlyTracked(action));
+  base::AutoLock scoped_lock(mutex_);
+
+  auto duration = clock_->NowTicks() - tracked_actions_start_times_[action];
+  auto duration_per_sample = duration / number_of_samples;
```

Comment:
Should `number_of_samples` be asserted to be non-zero?

Reply:
Fixed.